docs/site/Express-middleware.md

```diff
@@ -0,0 +1,319 @@
+   point/extension capabilities.
+
+## Bring Express middleware to LoopBack
+
+Let's start with a few examples to illustrate how we can bring Express
+middleware to LoopBack applications with minimal efforts.
+
+### Use Express middleware within the sequence of actions
+
+Express middleware can now be plugged into the REST sequence with an
+`InvokeMiddleware` action being injected to the default sequence.
+
+#### Invoke Express middleware explicitly in the sequence
+
+The custom sequence class below invokes two Express middleware
+([`helmet`](https://helmetjs.github.io/) and
+[`morgan`](https://github.com/expressjs/morgan)) handler functions as the first
+step.
+
+{% include code-caption.html content="src/sequence.ts" %}
+
+```ts
+import helmet from 'helmet'; // For security
+import morgan from 'morgan'; // For http access logging
+
+export class MySequence extends DefaultSequence {
+  async handle(context: RequestContext): Promise<void> {
+    try {
+      const {request, response} = context;
+      // `this.invokeMiddleware` is an injected function to invoke a list of
+      // Express middleware handler functions
+      const finished = await this.invokeMiddleware(context, [
+        helmet({}), // options for helmet is fixed and cannot be changed at runtime
+        morgan('combined', {}), // options for morgan is fixed and cannot be changed at runtime
+      ]);
+      if (finished) {
+        // The http response has already been produced by one of the Express
+        // middleware. We should not call further actions.
+        return;
+      }
+      const route = this.findRoute(request);
+      const args = await this.parseParams(request, route);
+      const result = await this.invoke(route, args);
+
+      this.send(response, result);
+    } catch (error) {
+      this.reject(context, error);
+    }
+  }
+}
+```
+
+#### Register Express middleware to be executed by `InvokeMiddleware` actions
+
+While the explicit Express middleware invocation is easy and simple, there are
+some limitations.
+
+1. The list of Express middleware has to be hard-coded in `src/sequence.ts`.
+   It's not easy to plug in a new middleware.
+2. The configuration of Express middleware is fixed unless we use dependency
+   injections for such values in `src/sequence.ts`.
+
+We provide another option to make invocation of Express middleware more flexible
+and extensible. The `InvokeMiddleware` actions within the sequence can discover
+registered middleware and invoke them in a chain.
+
+We first register middleware against the default or a named chain using APIs
+from `RestApplication`. It can happen in the constructor of an application.
+
+{% include code-caption.html content="src/application.ts" %}
+
+```ts
+import morgan from 'morgan';
+import {ApplicationConfig} from '@loopback/core';
+import {RestApplication} from '@loopback/rest';
+
+export class MyApplication extends RestApplication {
+  constructor(config: ApplicationConfig) {
+    this.expressMiddleware(
```

Comment:
Maybe `useExpressMiddleware` or `useMiddleware` would be a better name? The usual convention for method names is to start with a verb.

Reply:
I choose to use `expressMiddleware` to be consistent with other registration related apis, such as:

- app.controller()
- app.component()
- app.dataSource()
- app.service()